src/cpp/session/modules/NotebookData.R

```diff
@@ -19,7 +19,20 @@
     "print.data.frame" = function(x, options) list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
     "print.tbl_df" = function(x, options)     list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
     "print.grouped_df" = function(x, options) list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
-    "print.data.table" = function(x, options) list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
+    "print.data.table" = function(x, options) {
+      shouldPrintTable <- TRUE
+      if (exists(x = "shouldPrint", envir = as.environment("package:data.table"))) {
```

Comment:
We can also check here for package version since this check is getting a bit more complicated that I would want to.